odoo/http.py

```diff
@@ -415,6 +368,133 @@ def serialize_exception(exception):
+            mimetype=attachment.mimetype,
+            as_attachment=True,
+            download_name=attachment.name,
+            conditional=True,
+            etag=attachment.checksum,
+            max_age=STATIC_CACHE_LONG,
+        )
+
+        if attachment.stored_fname:
+            stat = os.stat
+            self.type = 'path'
+            self.path = werkzeug.security.safe_join(
+                os.path.abspath(config.filestore()),
+                attachment.stored_fname
+            )
+            self.last_modified = stat.mtime
+            self.size = stat.size
+
+        elif attachment.db_datas:
+            self.type = 'data'
+            self.data = attachment.raw
+            self.size = len(self.data)
+
+        elif attachment.url:
+            self.type = 'url'
+            self.url = attachment.url
+
+            # When the URL targets a file located in an addon, assume it
+            # is a path instead. It saves a 3xx response.
+            if (attachment.url or '').startswith('/'):
+                with contextlib.suppress(FileNotFoundError):
+                    return cls.from_path(attachment.url)
+
+        return self
+
+    @classmethod
+    def from_image_attachment(cls, attachment, resized_image_data=None):
+        self = cls.from_attachment(attachment)
+        self.as_attachment = False
+        self.etag = f'{attachment.checksum}-{self.size}'
+
+        if resized_image_data:
+            self.type = 'data'
+            self.path = None
+            self.data = resized_image_data
+            self.etag = f'{attachment.checksum}-{len(resized_image_data)}'
+
+    @wraps(werkzeug_send_file)
+    def make_http_response(self, **send_file_kwargs):
+        assert self.type in ('url', 'data', 'path') and getattr(self, self.type) is not None
+
+        if self.type == 'url:
+            return request.redirect(self.url)
+
+        for attr in SEND_FILE_ATTRS:
+            send_file_kwargs.setdefault(getattr(self, attr))
+
+        if self.type == 'data':
+            if len(content) > LARGE_FILE_THRESHOLD:
+                _logger.warning(LARGE_FILE_WARNING, self.download_name, self.size // (1 << 20))
+            return werkzeug.utils.send_file(
+                io.BinaryIO(self.data),
+                request.httprequest.environ,
+                use_x_sendfile=False,
+                response_class=Response,
+                **send_file_kwargs,
+            )
+
+        # self.type == 'path'
+        if self.size > LARGE_FILE_THRESHOLD and not config['x_accel']:
+            _logger.warning(MISSING_CSRF_WARNING, self.path, self.size // (1 << 20))
+
+        res = werkzeug.utils.send_file(
+            self.path,
+            request.httprequest.environ,
+            use_x_sendfile=config['x_accel'],
+            response_class=Response,
+            **send_file_kwargs,
+        )
+
+        if 'x-sendfile' in res.headers:
+            res.headers['x-accel-redirect'] = res.headers['x-sendfile']
```

Comment:
The semantics are completely different tho, `Sendfile` takes a filesystem path while `Accel-Redirect` takes a URI. That's why [it was never supported in flask](https://github.com/pallets/flask/issues/72) (and probably why it's not supported in werkzeug). So the header's value has to be completely rewritten to match the expected location.

Reply:
fixed